mapclassify/util.py

```diff
@@ -0,0 +1,73 @@
+from ._classify_API import classify
+
+
+def get_rgba(
+    values,
+    classifier="quantiles",
+    k=6,
+    cmap="viridis",
+    alpha=1,
+    nan_color=[255, 255, 255, 255],
+):
+    """Convert array of values into RGBA colors using a colormap and classifier.
+
+    Parameters
+    ----------
+    values : list-like
+        array of input values
+    classifier : str, optional
+        string description of a mapclassify classifier, by default "quantiles"
+    k : int, optional
+        number of classes to form, by default 6
+    cmap : str, optional
+        name of matplotlib colormap to use, by default "viridis"
+    alpha : float
+        alpha parameter that defines transparency. Should be in the range [0,1]
+    nan_color : list, optional
+        RGBA color to fill NaN values, by default [255, 255, 255, 255]
+
+    Returns
+    -------
+    numpy.array
+        array of lists with each list containing four values that define a color using
+        RGBA specification.
+    """
+    try:
+        import pandas as pd
+        from matplotlib import cm
+        from matplotlib.colors import Normalize
+    except ImportError as e:
+        raise ImportError("This function requires pandas and matplotlib") from e
+    if not (alpha <= 1) and (alpha >= 0):
+        raise ValueError("alpha must be in the range [0,1]")
+    if not pd.api.types.is_list_like(nan_color) and not len(nan_color) == 4:
+        raise ValueError("`nan_color` must be list-like of 4 values: (R,G,B,A)")
+
+    # only operate on non-NaN values
+    v = pd.Series(values, dtype=object)
+    legit_indices = v[~v.isna()].index.values
+
+    # transform (non-NaN) values into class bins
+    bins = classify(v.dropna().values, scheme=classifier, k=k).yb
```

Comment:
Shall we not require `k` and pass through all `kwargs` to support all classifiers and their options?

Reply:
oh, dur. good point